scripts/linter/index.js

```diff
@@ -0,0 +1,49 @@
+const frontmatter = require("remark-frontmatter");
+const parse = require("remark-parse");
+const report = require("vfile-reporter");
+const stringify = require("remark-stringify");
+const unified = require("unified");
+const vfile = require("to-vfile");
+
+const collectRecipes = require("./collect-recipes");
+const collectDocs = require("./collect-docs");
+const deprecatedSections = require("./plugins/deprecated-sections");
+const missingSections = require("./plugins/missing-sections");
+const slugifySections = require("./plugins/slugify-sections");
+const validRecipe = require("./plugins/valid-recipes");
+const yamlLoader = require("./plugins/yaml-loader");
+
+async function main() {
+    const recipes = await collectRecipes();
+
+    const markdownParser = unified()
+        .use(parse)
+        .use(stringify)
+        .use(frontmatter, ["yaml"])
+        .use(yamlLoader)
+        .use(validRecipe, { recipes })
+        .use(slugifySections)
+        .use(deprecatedSections, {
+            sections: {
+                "html-element": ["usage_notes"],
+                "css-property": ["usage_notes"]
+            }
+        })
+        .use(missingSections);
+
+    const filePaths = await collectDocs();
+
+    const reportedFiles = [];
+
+    for (const fp of filePaths) {
+        const file = await vfile.read(fp);
+        await markdownParser().process(file);
+        reportedFiles.push(file);
+    }
+
+    console.error(report(reportedFiles, { quiet: true }));
```

Comment:
```
▶ npm run lint-md

> stump@0.1.0 lint-md /Users/peterbe/dev/MOZILLA/MDN/stumptown-content
> node scripts/linter

content/css/reference/properties/border/border.md
            warning  "css-property" recipes requires section "example-syntax"     missing-section           stumptown-linter
            warning  "css-property" recipes requires section "see-also"           missing-section           stumptown-linter
            warning  "css-property" recipes requires section "short-description"  missing-section           stumptown-linter
            warning  "css-property" recipes requires section "syntax-overview"    missing-section           stumptown-linter

content/css/reference/properties/border-block-start-width/border-block-start-width.md
            warning  "css-property" recipes requires section "example-syntax"     missing-section           stumptown-linter
            warning  "css-property" recipes requires section "see-also"           missing-section           stumptown-linter
            warning  "css-property" recipes requires section "short-description"  missing-section           stumptown-linter
            warning  "css-property" recipes requires section "syntax-overview"    missing-section           stumptown-linter

content/css/reference/properties/transform/transform.md
            warning  "css-property" recipes requires section "example-syntax"     missing-section           stumptown-linter
            warning  "css-property" recipes requires section "see-also"           missing-section           stumptown-linter
            warning  "css-property" recipes requires section "short-description"  missing-section           stumptown-linter
            warning  "css-property" recipes requires section "syntax-overview"    missing-section           stumptown-linter

content/html/HTML.md
  1:1-14:4  error    "landing-page" recipe was not found in /recipes              frontmatter-valid-recipe  stumptown-linter

content/html/guides/Applying_color/Applying_color.md
   1:1-6:4  error    "guide" recipe was not found in /recipes                     frontmatter-valid-recipe  stumptown-linter

content/html/reference/elements/base/base.md
            warning  "html-element" recipes requires section "see_also"           missing-section           stumptown-linter

content/html/reference/elements/datalist/datalist.md
            warning  "html-element" recipes requires section "overview"           missing-section           stumptown-linter

content/html/reference/elements/dd/dd.md
            warning  "html-element" recipes requires section "overview"           missing-section           stumptown-linter

content/html/reference/elements/elements.md
   1:1-9:4  error    "landing-page" recipe was not found in /recipes              frontmatter-valid-recipe  stumptown-linter

content/html/reference/elements/figcaption/figcaption.md
            warning  "html-element" recipes requires section "overview"           missing-section           stumptown-linter

content/html/reference/elements/html/html.md
            warning  "html-element" recipes requires section "overview"           missing-section           stumptown-linter

content/html/reference/elements/iframe/iframe.md
            warning  "html-element" recipes requires section "see_also"           missing-section           stumptown-linter

content/html/reference/elements/legend/legend.md
            warning  "html-element" recipes requires section "overview"           missing-section           stumptown-linter

content/html/reference/elements/table/table.md
            warning  "html-element" recipes requires section "overview"           missing-section           stumptown-linter

content/learn/html/Introduction_to_HTML/Advanced_text_formatting/Advanced_text_formatting.md
   1:1-6:4  error    "guide" recipe was not found in /recipes                     frontmatter-valid-recipe  stumptown-linter

content/learn/html/Introduction_to_HTML/Creating_hyperlinks/Creating_hyperlinks.md
   1:1-6:4  error    "guide" recipe was not found in /recipes                     frontmatter-valid-recipe  stumptown-linter

content/learn/html/Introduction_to_HTML/Debugging_HTML/Debugging_HTML.md
   1:1-6:4  error    "guide" recipe was not found in /recipes                     frontmatter-valid-recipe  stumptown-linter

content/learn/html/Introduction_to_HTML/Document_and_website_structure/Document_and_website_structure.md
   1:1-6:4  error    "guide" recipe was not found in /recipes                     frontmatter-valid-recipe  stumptown-linter

content/learn/html/Introduction_to_HTML/Getting_started/Getting_started.md
   1:1-6:4  error    "guide" recipe was not found in /recipes                     frontmatter-valid-recipe  stumptown-linter

content/learn/html/Introduction_to_HTML/HTML_text_fundamentals/HTML_text_fundamentals.md
   1:1-6:4  error    "guide" recipe was not found in /recipes                     frontmatter-valid-recipe  stumptown-linter

content/learn/html/Introduction_to_HTML/Introduction_to_HTML.md
   1:1-6:4  error    "guide" recipe was not found in /recipes                     frontmatter-valid-recipe  stumptown-linter

content/learn/html/Introduction_to_HTML/Marking_up_a_letter/Marking_up_a_letter.md
   1:1-6:4  error    "guide" recipe was not found in /recipes                     frontmatter-valid-recipe  stumptown-linter

content/learn/html/Introduction_to_HTML/Structuring_a_page_of_content/Structuring_a_page_of_content.md
   1:1-6:4  error    "guide" recipe was not found in /recipes                     frontmatter-valid-recipe  stumptown-linter

content/learn/html/Introduction_to_HTML/The_head_metadata_in_HTML/The_head_metadata_in_HTML.md
   1:1-6:4  error    "guide" recipe was not found in /recipes                     frontmatter-valid-recipe  stumptown-linter

33 messages (✖ 13 errors, ⚠ 20 warnings)

▶ echo $?
0
```

So even though there were 13 errors it still finished OK. 
I'm not really sure how `vfile-reporter` works (or even what it is!) but perhaps we leave that and on the line below this one, we filter `reportedFiles` for things that are actual errors and do something like `process.exitCode = reportedFiles.filter(v => v.error).length;`

Reply:
[`vfile`](https://github.com/vfile/vfile) is `unified`'s library for doing stuff with files. It's pretty cool. I'm using it here for attaching (and pretty printing) error messages, though it does other things (for example, tracks file renames). It also provides an API for getting stats about those reports, which I've used to set the exit code. Thank you!